hbw/app/javascript/packs/hbw/components/shared/context/connection.tsx

```diff
@@ -17,13 +20,48 @@ type InitParams = {
 }
 
 export const withConnectionContext = (
-  { connection }: InitParams
+  { connection }: InitParams,
 ) => (WrappedComponent: ComponentType) => {
-  const ConnectionProvider: FC = props => (
-      <ConnectionContext.Provider value={connection} >
+  const ConnectionProvider: FC = (props) => {
+    const [connectionError, setConnectionError] = useState(() => ({
+      exist:  false,
+      header: '',
+      body:   ''
+    }));
+
+    const handleError = async (response: Response) => {
+      if (response.status === 504) {
+        setConnectionError({
+          exist:  true,
+          header: 'errors.camunda_is_not_available',
+          body:   await response.json()
+        });
+      } else {
+        setConnectionError({
+          exist:  false,
+          header: '',
+          body:   ''
+        });
+      }
```

Comment:
What about 502 or 503 for example? Why it is not a connection error? 
I see you always return 503 from the backend but think it's a questionable design.

Reply:
Maybe this is overkill, and 504 is enough for proxying an error?